CRM/Contribute/Form/Search.php

```diff
@@ -390,6 +390,11 @@ public function fixFormValues() {
       $this->_defaults['contribution_status_id'] = array($status => 1);
     }
 
+    $pcpid=CRM_Utils_Request::retrieve('pcpid', 'Positive', $this);
+    if ($pcpid) {
```

Comment:
For multiple values:
`
 $pcpid=CRM_Utils_Request::retrieve('pcpid', 'String', $this);
    if ($pcpid) {
      $n = 0;
      foreach ($pcpid as $pcpIdList) {
        $this->_formValues['contribution_pcp_made_through_id'][$n]=$pcpIdList;
        $n++;
      }
    }
`